### PR TITLE
Adding byproducts to daily feed totals

### PR DESCRIPTION
**Pull request by JoeWaddell**
Changes cherry-picked from #1226

Adds Fd_Category to NRC comp library, and adds the total kg of byproducts fed to the ration totals output.

### CODE REVIEW

**Review by jadamchick — APPROVED** [review]
Looks good and runs well!

**Comment by jadamchick** [comment]
As soon as you get a second set of eyes (@tomhuhh maybe, when you have the chance?) on those changes to feeds 135 and 137 in NASEM, I'll cherry pick the commits into the pilot testing branch and update that as part of PR #1246 .

File: input/data/feed/NRC_comp.csv [hunk 1]
```diff
@@ -1,135 +1,135 @@
...
+118,By-Product/Other,Conc,77.9,8.5,0,0,0.2,0.1,0.1,0,11.4,79.8,1.04,82.91,74.1,25.9,0,3.2,100,0.15,0.03,0.29,6.06,1.48,0,0.6,3.6,FALSE,FALSE,kg,44.717,100,0
+119,By-Product/Other,Conc,74.3,5.8,0,0,0.2,0.4,0.2,0,13.3,80.3,1.04,80.94,74.1,25.9,0,3.2,100,1,0.1,0.42,4.01,0.22,0,0.47,3.47,FALSE,FALSE,kg,4.077,100,0
+120,Energy Source,Conc,90,13.2,1.8,0.3,5.1,30,14.6,4.9,3.3,50.2,1.04,78.51,65.2,28.8,6,17.4,75,0.11,0.4,0.16,0.52,0.03,0,0.19,3.47,FALSE,FALSE,kg,43.919,100,0
+121,Grain Crop Forage,Forage,91.9,9.1,1.3,0.6,2.2,58,36.4,6.5,8.5,23.5,1,55.91,35,53.1,11.9,4.3,70,0.37,0.22,0.17,2.01,0.33,1.08,0.14,2.46,FALSE,FALSE,kg,3.884,100,0
+123,Grain Crop Forage,Forage,34.6,12.9,2.1,1,3.4,60.6,38.9,5.5,9.8,15.4,1,56.78,45.6,30.9,23.5,5.4,65,0.52,0.31,0.2,2.89,0.24,1.34,0.19,2.54,FALSE,TRUE,kg,3.183,100,0
+129,By-Product/Other,Conc,92.3,51.8,5.8,1.1,1.4,21.4,13.5,4.6,5.8,25.4,1,74.75,61.7,36.6,1.7,16.1,90,0.2,0.64,0.32,1.32,0.03,0.1,0.32,3.85,FALSE,FALSE,kg,6.388,100,0
+135,By-Product/Other,Conc,5.4,10.5,5.2,2.3,10.8,22.1,16.5,2.3,12.8,49,1,80.7,4.5,90.5,5,2,90,0.49,0.29,0.11,1.04,0.26,0.19,0.11,3.51,FALSE,FALSE,kg,57.49,100,0
```
Comment (tomhuhh) [inline input/data/feed/NRC_comp.csv]: This DM% should be 35.4 instead of 5.4 as far as I remember.

File: input/data/feed/NASEM_Comp_with_TDN.csv [hunk 2]
```diff
@@ -58,43 +58,44 @@ rufas_id,Name,Fd_Category,feed_type,DM,ash,CP,N_A,N_B,N_C,Kd,RUP,dRUP,sol_prot,A
...
+137,Poultry byproduct meal,By-Product/Other,Conc,95.521,14.247,65.624,5,90,5,2,70.608,90,28.234,3.89,30.34,0,0,0,0,0,0,11.779,12.779,4.252,4.314,2.477,0.161,0.893,0.376,0.554,0.742,14.139,233.366,44.761,121.565,1.196,0,0,0,0,4.3,1.38,2.5,4.5,3.9,1.3,2.5,2.5,0.7,3.1,0,0,0,0,0,0,0,0,0,0,0,91,73,0,100,0,0,0,0,0,0,0,0.6,0.68,1,0.92,1,0.05,0.1,0.31,0.004,0.2,FALSE,TRUE,kg,100,0,78.98,4.05
 138,Rice bran,By-Product/Other,Conc,92.376,10.021,14.813,31,45,24,3.38,53.429,85.033,30.483,0.778,2.18,13.84,23.055,23.333,4.943,22.181,7.953,12,18.503,3.21,0.779,1.774,0.784,1.402,0.02,0.133,0.174,8.645,215.993,182.668,60.923,1.139,0,0,0,0.17,7.739,2.75,3.756,7.141,4.727,2.181,4.446,3.88,1.218,5.719,0,0.28,17.53,0.21,1.56,0,39.1,38.11,1.46,1.75,0,91,73,5,95,0,0,0,0,0,0,0,0.6,0.688,1,0.92,1,0.05,0.1,0.31,0.004,0.2,FALSE,TRUE,kg,100,0,84.77,3.76
```
Comment (tomhuhh) [inline input/data/feed/NASEM_Comp_with_TDN.csv:69]: I would also just double check that the assumed TDN and DE (from 114 meat meal) are correctly entered.

**Review by tomhuhh — CHANGES_REQUESTED** [review]

File: input/data/feed/NRC_comp.csv (hunk 1 above)
Comment (JoeWaddell) [reply input/data/feed/NRC_comp.csv]: Corrected, thanks!

**Review by tomhuhh — APPROVED** [review]
LGTM!

**Review by allisterakun — APPROVED** [review]
Code looks good! Straight-forward changes.

**Pull request merged**